src/main/scala/org/apache/spark/sql/kinesis/KinesisSourceProvider.scala

```diff
@@ -241,17 +241,17 @@ private[kinesis] object KinesisSourceProvider extends Logging {
 
 
   private[kinesis] def getKinesisPosition(
-      params: Map[String, String]): KinesisPosition = {
-    // TODO Support custom shards positions
+      params: Map[String, String]): InitialKinesisPosition = {
     val CURRENT_TIMESTAMP = System.currentTimeMillis
     params.get(STARTING_POSITION_KEY).map(_.trim) match {
       case Some(position) if position.toLowerCase(Locale.ROOT) == "latest" =>
-        new AtTimeStamp(CURRENT_TIMESTAMP)
+        InitialKinesisPosition.fromPredefPosition(new AtTimeStamp(CURRENT_TIMESTAMP))
       case Some(position) if position.toLowerCase(Locale.ROOT) == "trim_horizon" =>
-        new TrimHorizon
+        InitialKinesisPosition.fromPredefPosition(new TrimHorizon)
       case Some(position) if position.toLowerCase(Locale.ROOT) == "earliest" =>
-        new TrimHorizon
-      case None => new AtTimeStamp(CURRENT_TIMESTAMP)
+        InitialKinesisPosition.fromPredefPosition(new TrimHorizon)
+      case Some(json) => InitialKinesisPosition.fromCheckpointJson(json)
```

Comment:
instead shall we use `InitialKinesisPosition.fromCheckpointJson(json, new AtTimeStamp(CURRENT_TIMESTAMP))`